docker/daemon/client.go

```diff
@@ -23,7 +18,6 @@ func newDockerClient(sys *types.SystemContext) (*dockerclient.Client, error) {
 
 	opts := []dockerclient.Opt{
 		dockerclient.WithHost(host),
-		dockerclient.WithVersion(defaultAPIVersion),
```

Comment:
AFAICT this needs to be replaced with `dockerclient.WithAPIVersionNegotiation()` to _actually_ opt into negotiating the version.

(The effect of negotiating is that the client and server use the lower of the two versions, agreeing on a mutually acceptable one.)

Without that opt-in, the client uses version-unspecified APIs, which the server interprets as “the current API”, i.e. every upgrade of the server can change semantics.

Can you make that change, and double-check that everything works as expected, please?

Reply:
Ah good catch I did test with my current code and it worked but having looked again, without the change you suggested, we would just get the `api.DefaultVersion` from the docker client code which is currently 1.44. Which would mean we would fail against older daemons

Reply:
> Without that opt-in, the client uses version-unspecified APIs, which the server interprets as “the current API”, i.e. every upgrade of the server can change semantics.

I was incorrect about this. As you say, without negotiation the client uses the latest known version (at time of vendoring of the client), which right now means it would fail against Docker ≤24:
```
Error response from daemon: client version 1.44 is too new. Maximum supported API version is 1.43"
```

So, yes, we do _need_ the version negotiation opt-in.